### PR TITLE
LSIF: RFC127-4: Some light API cleanup

### PR DESCRIPTION
**Pull request by efritz**
This PR handles some misc cleanup, notably it changes the API introduced in https://github.com/sourcegraph/sourcegraph/pull/9179 so that the dump manager is not returning entire document data payloads (which can be sizeable). Instead it returns monikers attached to a range (which is what the server was using the document for in the first place), or returns package information by ID.

This also takes the opportunity to remove some code that was there to support leaky integration test that no longer exist.

### CODE REVIEW

File: doc/dev/architecture/architecture.svg [hunk 1]
```diff
@@ -2,569 +2,603 @@
 <?xml-stylesheet href="/assets/architecture.css" type="text/css"?>
```
Comment (felixfbecker) [inline doc/dev/architecture/architecture.svg]: It seems like this reformatted the graph to no longer have the control flow read top to bottom (clients -> frontend -> ...). Could that be maintained?

File: lsif/src/dump-manager/settings.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@ export const HTTP_PORT = readEnvInt('HTTP_PORT', 3187)
 /** HTTP address for internal LSIF HTTP API. */
 export const LSIF_SERVER_URL = process.env.LSIF_SERVER_URL || 'http://lsif-server'
 
-/** Where on the file system to store LSIF files. */
+/** Where on the file system to store LSIF files. This is assumed to be a persistent volume. */
```
Comment (creachadair) [inline lsif/src/dump-manager/settings.ts]:
```suggestion
/** Where on the file system to store LSIF files. This should be a persistent volume. */
```

File: doc/dev/architecture/architecture.svg (hunk 1 above)
Comment (efritz) [reply doc/dev/architecture/architecture.svg]: Dot sucks.

File: lsif/docs/manager.yaml [hunk 3]
```diff
@@ -0,0 +1,460 @@
...
+        character:
+          type: number
+          description: The (zero-index) character index.
+      required:
+        - line
+        - character
+      additionalProperties: false
+    Range:
+      type: object
+      description: A closed range of positions in a source file.
```
[the remark this answers is not in the document]
Comment (efritz) [reply lsif/docs/manager.yaml]: Updated.

File: doc/dev/architecture/architecture.svg (hunk 1 above)
Comment (efritz) [reply doc/dev/architecture/architecture.svg]: Tried to update.

**Pull request merged**